### PR TITLE
HUBS-1842 New Delphix Plugin commit to develop

### PR DESCRIPTION
**Pull request by ankit-patil-hubs**
### Context
This PR introduces a redesigned and rearchitected Jenkins plugin for Delphix. 

### Solution
- Consuming the DCT java SDK.
- Support for provision VDB operation.
- Support for delete VDB operation.

### Testing Done
Tested via running the plugin in local host

### JIRA
https://delphix.atlassian.net/browse/HUBS-1842

### CODE REVIEW

File: src/main/webapp/help-credentialId.html [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<div>
+    Specify the Api key to be used for this operation
```
[the remark this answers is not in the document]
Comment (nick-mathison) [reply src/main/webapp/help-credentialId.html:2]: I forget if it's `apk` pre-pended or not, but we should give an example.

(That actually might be a nice QoL add, to append `apk` if it's not found or remove it if it is.)

File: src/main/webapp/help-jsonParam.html [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<div>
+    (Optional) Specify any additional parametrs in JSON format.
+    <strong> Note:- If the same parameter from above form is provided in JSON the form parameter takes preference and
```
Comment (nick-mathison) [inline src/main/webapp/help-jsonParam.html:3]: `Note:- If the same parameter is specified in another field, that other value will take precedence.`

File: src/main/webapp/help-name.html [hunk 3]
```diff
@@ -0,0 +1,3 @@
+<div>
+    (Optional) The unique name of the provisioned VDB within a group. If unset, a name is randomly generated
```
Comment (nick-mathison) [inline src/main/webapp/help-name.html:2]: I don't think the `within a group.` is needed? Not all VDBs are part of a group?

File: src/main/webapp/help-optional.html [hunk 4]
```diff
@@ -0,0 +1,3 @@
+<div>
+    (Optional) Additional optional parameters that can be used for provision operation
```
[the remark this answers is not in the document]
Comment (nick-mathison) [reply src/main/webapp/help-optional.html:2]: Also, a Jenkins "job" is roughly equivalent to the complete `Jenkinsfile` or a collection of multiple steps.

File: src/main/webapp/help-skipPolling.html [hunk 5]
```diff
@@ -0,0 +1,5 @@
+<div>
+    (Optional) If checked the jenkins job will not wait for the provison operation to complete
+    <strong> Note:- The newly provisioned VDB would be in UNKNOWN state
```
Comment (nick-mathison) [inline src/main/webapp/help-skipPolling.html:3]: `...will be in a temporary UNKNOWN state.`

Comment (nick-mathison) [reply src/main/webapp/help-skipPolling.html:3]: (Optional) If checked, the Jenkins step will not wait for the provisioning to complete.

File: src/main/webapp/help-snapshotId.html [hunk 6]
```diff
@@ -0,0 +1,4 @@
+<div>
+    The ID or name of the snapshot from which to execute the operation. If the snapshot_id is not, selects the latest
+    snapshot
```
Comment (nick-mathison) [inline src/main/webapp/help-snapshotId.html:3]: The ID or name of the Snapshot from which to execute the step. If the snapshot_id is not specified, then DCT will automatically select the latest Snapshot.

File: src/main/webapp/help-tagList.html [hunk 7]
```diff
@@ -0,0 +1,4 @@
+<div>
+    (Optional) The tags to be created for VDB. This is a map of 2 parameters:
```
Comment (nick-mathison) [inline src/main/webapp/help-tagList.html:2]: `{Optional) The tags to be assigned to the VDB.`

File: src/main/webapp/help-vdbRestart.html [hunk 8]
```diff
@@ -0,0 +1,3 @@
+<div>
+    (Optional) If checked the VDB would restart after provision is complete
```
Comment (nick-mathison) [inline src/main/webapp/help-vdbRestart.html:2]: `(Optional) If checked the VDB will restart after provisioning is complete.`

**Review by nick-mathison — CHANGES_REQUESTED** [review]

File: src/main/resources/io/jenkins/plugins/delphix/Messages.properties [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+
+
+ProvisionVDBSnapshot.DisplayName=Delphix: Provision VDB From Snapshot
+ProvisionVDBSnapshot.Info=Provison VDB From Snapshot Operation Started with BuildID: {0}
+
+
+ProvisionVDB.Complete=VDB Provision Job Completed
+ProvisionVDB.Fail=VDB Provision Job Failed
+ProvisionVDB.Save=Saving Results to File {0}
+ProvisionVDB.Start=Provison Started For Target VDBID: {0} with JobID: {1}
```
[the remark this answers is not in the document]
Comment (ankit-patil-hubs) [reply src/main/resources/io/jenkins/plugins/delphix/Messages.properties:34]: I was not sure what works best , we can keep VDBID or VDB ID or VDB_ID

File: src/main/webapp/help-credentialId.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (ankit-patil-hubs) [reply src/main/webapp/help-credentialId.html:2]: apk is not pre-pended , I will add the dct  documentation link for the same

File: src/main/webapp/help-dctUrl.html [hunk 10]
```diff
@@ -0,0 +1,4 @@
+<div>
+    The host name for where DCT APIGW is running.
+    <strong> The connection URL must take the following format: https://{DCT_HOSTNAME}/{api_version}</strong>
```
[the remark this answers is not in the document]
Comment (ankit-patil-hubs) [reply src/main/webapp/help-dctUrl.html:3]: leys keep connection URL for both

File: src/main/resources/io/jenkins/plugins/delphix/Messages.properties [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+Apiclient.Fail=ApiClient Creation Failed
+VDBId.Empty=Please Set a VDB ID
+#SnapshotId.Empty=Please Set a Snapshot ID
+BookmarkId.Empty=Please Set a Bookmark ID
+Credential.Empty=Please Select a Credential ID
+
+Json.Invalid=Invalid Json Format
+Json.IncorrectKey=Invalid Provision Parameter {0}
+
+
```
[the remark this answers is not in the document]
Comment (ankit-patil-hubs) [reply src/main/resources/io/jenkins/plugins/delphix/Messages.properties:11]: got it

File: src/main/webapp/help-environmentUserId.html [hunk 12]
```diff
@@ -0,0 +1,3 @@
+<div>
+    (Optional)[Updatable] The environment user ID to use to connect to the target environment
```
[the remark this answers is not in the document]
Comment (ankit-patil-hubs) [reply src/main/webapp/help-environmentUserId.html:2]: it seems to be a typo from terraform help page

File: src/main/webapp/help-name.html (hunk 3 above)
Comment (ankit-patil-hubs) [reply src/main/webapp/help-name.html:2]: yes this was a bit confusing, this message is the same in Terraform as well, https://registry.terraform.io/providers/delphix-integrations/delphix/latest/docs/resources/vdb#argument-reference 

we can change if this is not the case

File: README.md [hunk 13]
```diff
@@ -22,52 +22,36 @@ The Delphix Plugin allows Jenkins to connect to multiple Delphix Engines and exe
...
+After the plugin has been installed, DCT will need to be connected. Go to System Configuration page, scroll down to the Delphix section. Enter your DCT URL.
 
-#### Self Service Container
+#### Credentials
 
-To interact with Self Service Containers, select "Delphix - Self Service Container" from the "Add Build Step" drop down in the Build phase section. Select an engine that was configured in the global configuration, select one of the operations listed below. Depending on the operation selected, it will either depend on a specified Self Service Container or a Bookmark. Read more about [Delphix Self Service Containers](https://docs.delphix.com/docs/delphix-self-service/delphix-self-service-admin-guide/understanding-data-containers).
+The API KEY has to be saved on Jenkins’s side as a Secret text , In Jenkins, go to Credentials > global > Add Credentials, select the Secret text type and fill in the inputs.
```
Comment (nick-mathison) [inline README.md:43]: Update Line: 
`The API KEY has to be saved on Jenkins’s side as a Secret text. In Jenkins, go to Credentials > Global > Add Credentials, select the Secret text type, and fill in the inputs.`

File: README.md [hunk 14]
```diff
@@ -4,7 +4,7 @@
 [![Jenkins Plugins](https://img.shields.io/jenkins/plugin/v/delphix.svg)](https://plugins.jenkins.io/delphix)
 [![License](https://img.shields.io/github/license/jenkinsci/delphix-plugin.svg)](LICENSE)
 
-The Delphix Plugin allows Jenkins to connect to multiple Delphix Engines and execute operations on Self Service Containers and Self Service Bookmarks.
+The Delphix Plugin allows Jenkins to connect to DCT and execute operations.
```
Comment (nick-mathison) [inline README.md:7]: Update DCT to `Data Control Tower (DCT)`

Comment (nick-mathison) [reply README.md:7]: lowercase `plugin`

File: src/main/resources/io/jenkins/plugins/delphix/Messages.properties (hunk 9 above)
[the remark this answers is not in the document]
Comment (nick-mathison) [reply src/main/resources/io/jenkins/plugins/delphix/Messages.properties]: When I see "VDBID" or "VDB_ID", I think you are referring to the variable. This should match the variable used in the Jenkins pipeline script calls. You wouldn't see these being used in our documentation unless it's referencing the direct CLI parameter. This idea could work in this context.

If I see "VDB ID", I'd think you are referring to its official name. This makes more sense in a general description of what's going wrong.

File: src/main/resources/io/jenkins/plugins/delphix/Messages.properties [hunk 15]
```diff
@@ -0,0 +1,50 @@
...
+ProvisionVDBSnapshot.Info=Provison VDB from snapshot operation started with BuildID: {0}
+
+
+ProvisionVDB.Complete=Provision VDB job completed
+ProvisionVDB.Fail=Provision VDB job failed
+ProvisionVDB.Save=Saving results to File: {0}
+ProvisionVDB.Start=Provison started for Target VDBID: {0} with JobID: {1}
+
+ProvisionVDBBookmark.DisplayName=Delphix: Provision VDB From Bookmark
+ProvisionVDBBookmark.Info=Provison VDB from bookmark operation started with buildID: {0}
```
Comment (nick-mathison) [inline src/main/resources/io/jenkins/plugins/delphix/Messages.properties:38]: Above capitalized the B in build. I'd expect it capitalized here too.

Comment (nick-mathison) [reply src/main/resources/io/jenkins/plugins/delphix/Messages.properties:38]: "Provision VDB From Bookmark" should be capitalized.

File: src/main/resources/io/jenkins/plugins/delphix/Messages.properties [hunk 16]
```diff
@@ -0,0 +1,50 @@
...
+Delete.Message5=Deleting VDB with name: {0}
+Delete.Complete=Delete VBD job completed
+Delete.Fail=Delete VDB job failed
+Delete.Error1=VDBID or Name is required
+Delete.Error2=More than 1 VDB found with the name: {0}
+Delete.Error3=No VDB found with the name: {0}
+
+
+ProvisionVDBSnapshot.DisplayName=Delphix: Provision VDB From Snapshot
+ProvisionVDBSnapshot.Info=Provison VDB from snapshot operation started with BuildID: {0}
```
Comment (nick-mathison) [inline src/main/resources/io/jenkins/plugins/delphix/Messages.properties:29]: Capitalize the operation name.

File: src/main/webapp/help-name.html (hunk 3 above)
Comment (nick-mathison) [reply src/main/webapp/help-name.html]: Based on my experience, this is not the case.

**Review by nick-mathison — CHANGES_REQUESTED** [review]

File: src/main/resources/index.jelly [hunk 17]
```diff
@@ -1,4 +1,5 @@
 <?jelly escape-by-default='true'?>
 <div>
-    Delphix
-</div>
+     Allow Jenkins Users to create and delete Delphix Objects.
```
Comment (nick-mathison) [inline src/main/resources/index.jelly:3]: Please update: 
The Delphix Plugin enables application teams to integrate the power of Delphix's DevOps Data Platform within their automated Jenkins Pipelines to enhance DevOps Test Data Management practices.

**Review by arunskurian — APPROVED** [review]

**Review by nick-mathison — APPROVED** [review]

**Pull request merged**